### PR TITLE
fix: properly transform invalid identifiers

### PR DESCRIPTION
**Pull request by paoloricciuti**
This fix the transform to handle the case where the template name is an invalid identifier.

Before it was using `camelCase` (only for the `Story` actually so that was also a bug) but the problem with `camelCase` is that it can lead to the same name from different string. For example `Modal && Trigger` and `Modal || Trigger` have the same `camelCase`.

To fix this i used a small hash function (stolen from the svelte code base) so that if the template is not a valid identifier (which needs to be to be a `Snippet`) it convert it to something like `template_skajsh` (which we don't care too much except for stack traces).

I don't know if it's a solution that you like but it works.

### CODE REVIEW

**Comment by xeho91** [comment]
🧡 

Apologies for missing the case where the transformation of the legacy template component `id` needs to be a valid identifier for snippet.

---

BTW, Jeppe. We could implement some formatting CI to this repo _(Prettier probably is the best choice since it supports Svelte, unlike Biome)_.

File: src/compiler/pre-transform/codemods/legacy-story.ts [hunk 1]
```diff
@@ -11,6 +9,7 @@ import {
 } from '$lib/parser/ast.js';
 import { InvalidTemplateAttribute } from '$lib/utils/error/legacy-api/index.js';
 
+import { hashTemplateName } from '#utils/identifier-utils';
```
Comment (JReinhold) [inline src/compiler/pre-transform/codemods/legacy-story.ts:12]:
```suggestion
import { hashTemplateName } from '$lib/utils/identifier-utils.js';
```

**Review by JReinhold — APPROVED** [review]
🙏

**Comment by JReinhold** [comment]
@paoloricciuti do you know why GH Actions aren't being runned in this PR? Did you disable it in your fork?

**Comment by paoloricciuti** [comment]
> @paoloricciuti do you know why GH Actions aren't being runned in this PR? Did you disable it in your fork?
> 
> (Gave you write-access to the repo so you don't need forks in the future)

Thanks @JReinhold ...no i did not disable anything at least not by choice...let me check

**Comment by paoloricciuti** [comment]
image

It should be ok...if you want i can redo the changes in a branch of the repo

**Comment by JReinhold** [comment]
> if you want i can redo the changes in a branch of the repo

Thanks that would be great, probably the easiest solution right now.

**Comment by paoloricciuti** [comment]
Closing in favor of #246

**Pull request closed without merging**